gql/parser.go

```diff
@@ -562,20 +580,182 @@ func parseArguments(l *lex.Lexer) (result []pair, rerr error) {
 	return result, nil
 }
 
+// String converts FilterTree to a string. Good for testing, debugging.
+func (t *FilterTree) String() string {
+	buf := bytes.NewBuffer(make([]byte, 0, 20))
+	t.stringHelper(buf)
+	return buf.String()
+}
+
+// stringHelper does simple DFS to convert FilterTree to string.
+func (t *FilterTree) stringHelper(buf *bytes.Buffer) {
+	if t == nil {
```

Comment:
This makes little sense. A function on t would crash, if t is nil, so this would never be called.